alvr/audio/src/lib.rs

```diff
@@ -57,47 +41,67 @@ fn device_from_custom_config(host: &Host, config: &CustomAudioDeviceConfig) -> R
     })
 }
 
-fn microphone_pair_from_sink_name(host: &Host, sink_name: &str) -> Result<(Device, Device)> {
-    let sink = host
-        .output_devices()?
-        .find(|d| d.name().unwrap_or_default().contains(sink_name))
-        .context("VB-CABLE or Voice Meeter not found. Please install or reinstall either one")?;
+#[allow(dead_code)]
+pub struct AudioDevice {
+    inner: Device,
+    is_output: bool,
+}
 
-    if let Some(source_name) = VIRTUAL_MICROPHONE_PAIRS.get(sink_name) {
-        Ok((
-            sink,
-            host.input_devices()?
-                .find(|d| {
-                    d.name()
-                        .map(|name| name.contains(source_name))
-                        .unwrap_or(false)
-                })
-                .context("Matching output microphone not found. Did you rename it?")?,
-        ))
+pub fn is_same_device(device1: &AudioDevice, device2: &AudioDevice) -> bool {
+    if let (Ok(name1), Ok(name2)) = (device1.inner.name(), device2.inner.name()) {
+        name1 == name2
     } else {
-        unreachable!("Invalid argument")
+        false
     }
 }
 
-#[allow(dead_code)]
-pub struct AudioDevice {
-    inner: Device,
-    is_output: bool,
+pub enum AudioRecordState {
+    Recording,
+    ShouldStop,
+    Err(Option<anyhow::Error>),
+}
+
+pub enum AudioChannel {
+    FrontLeft,
+    FrontRight,
+    Center,
+    SurroundLeft,
+    SurroundRight,
+    BackLeft,
+    BackRight,
+    Top,
+    HighFrontLeft,
+    HighFrontRight,
+    HighFrontCenter,
+    HighBackLeft,
+    HighBackRight,
+    LowFrequency,
 }
 
-#[cfg_attr(not(target_os = "linux"), allow(unused_variables))]
+macro_rules! channel_mix {
+    ( $x:expr ) => {
+        match $x {
+            AudioChannel::FrontLeft => [1.0, 0.0],
+            AudioChannel::FrontRight => [0.0, 1.0],
+            AudioChannel::Center => [0.707, 0.707],
+            AudioChannel::SurroundLeft => [0.707, 0.0],
+            AudioChannel::SurroundRight => [0.0, 0.707],
+            AudioChannel::BackLeft => [0.707, 0.0],
+            AudioChannel::BackRight => [0.0, 0.707],
+            AudioChannel::Top => [0.577, 0.577],
+            AudioChannel::HighFrontLeft => [0.707, 0.0],
+            AudioChannel::HighFrontRight => [0.0, 0.707],
+            AudioChannel::HighFrontCenter => [0.5, 0.5],
+            AudioChannel::HighBackLeft => [0.5, 0.0],
+            AudioChannel::HighBackRight => [0.0, 0.5],
+            _ => [0.0, 0.0],
+        }
+    };
+}
+
+#[cfg(target_os = "android")]
```

Comment:
Might want to move this out into it's own file if it's android only

Reply:
Plus as far as I can tell this code is also invoked on the non-android clients

Reply:
https://github.com/alvr-org/ALVR/blob/master/alvr/client_core/src/connection.rs#L327
As far as I can tell this code is relevant for all clients, not just android. If that's actually the case then you should rename that file from android to client and feature gate it as "not windows and not linux"

Reply:
Ehh I don't like that Idea of gating it behind Linux as there's a high chance of headsets reporting as Linux like the simula, is there a way to have custom platform targets?

Reply:
And besides that comment the platform specific audio seems to be all refactored out

Reply:
Out of that implementation yes, but it conflicts with the other impls for windows and linux. So we would probably need to gate behind something like "not windows and not pipewire" and just introduce a pipewire option.